config/checkstyle_checks.xml

```diff
@@ -399,6 +399,12 @@
       //LITERAL_CATCH//METHOD_CALL[.//IDENT[@text = 'printStackTrace']]/.."/>
       <message key="matchxpath.match" value="Avoid using 'printStackTrace'."/>
     </module>
+    <module name="MatchXpath">
+      <property name="query" value="//CLASS_DEF//OBJBLOCK//METHOD_DEF//MODIFIERS//
+            ANNOTATION[.//IDENT[@text='Test']]//ANNOTATION_MEMBER_VALUE_PAIR
+            [.//IDENT[@text='expected']]"/>
+      <message key="matchxpath.match" value="Avoid using ''expected'' in Test annotation."/>
```

Comment:
> ''expected'' in Test annotation

I would add what `expected` is after its wording. Like if expected was a class, it should say "avoid using expected class in ...". I assume expected is a member here based on AST name?

Reply:
Here is a good usage example from [SO](https://stackoverflow.com/questions/156503/how-do-you-assert-that-a-certain-exception-is-thrown-in-junit-4-tests):
```
@Test(expected = IndexOutOfBoundsException.class)
public void testIndexOutOfBoundsException() {

    ArrayList emptyList = new ArrayList();
    Object o = emptyList.get(0);

}
```

Reply:
So, to be more detailed, say: `Avoid using ''expected'' attribute in Test annotation.`

Reply:
I changed it to the above comment, let me know if this needs further improvement.